lib/url.js

```diff
@@ -22,48 +22,45 @@ function Url() {
   this.pathname = null;
   this.path = null;
   this.href = null;
+  this._prependSlash = false;
 }
 
 // Reference: RFC 3986, RFC 1808, RFC 2396
 
-// define these here so at least they only have to be
-// compiled once on the first module load.
-const protocolPattern = /^([a-z0-9.+-]+:)/i;
-const portPattern = /:[0-9]*$/;
-
-// Special case for a simple path URL
-const simplePathPattern = /^(\/\/?(?!\/)[^\?\s]*)(\?[^\s]*)?$/;
+const _protocolCharacters = makeAsciiTable([
+  [0x61, 0x7A]/*a-z*/,
+  [0x41, 0x5A]/*A-Z*/,
+  0x2E/*'.'*/, 0x2B/*'+'*/, 0x2D/*'-'*/
+]);
```

Comment:
Probably not; my understanding was that this is more or less a straight port of what Node already had, and any conformance updates would come later.